src/common/collection.c

```diff
@@ -470,57 +555,115 @@ gboolean dt_collection_get_sort_descending(const dt_collection_t *collection)
 gchar *dt_collection_get_sort_query(const dt_collection_t *collection)
 {
   gchar *sq = NULL;
+  gchar *second_order = NULL;/*remember previous sorting criteria as second order sorting criteria*/
```

Comment:
I don't see this released, so I suspect a memory leak.

Reply:
This is just local variable that keeps string for query. To avoid that it remains NULL, I just added default case which shouldn't occur anyway.

Reply:
But even so, you assign value to this variable:

`second_order` = dt_util_dstrcat(...);`

So you need to free this variable as dr_util_sstrcat() does allocate memory.

So when no more needed you need `g_free(second_order);`

Reply:
Oups, you're right. I'll fix this!